diff -u -r -N pology-0.12.ori/lang/ru/rules/all.rules pology-0.12/lang/ru/rules/all.rules
--- pology-0.12.ori/lang/ru/rules/all.rules	2014-07-24 22:57:17.000000000 +0400
+++ pology-0.12/lang/ru/rules/all.rules	2015-12-19 08:05:07.518346445 +0300
@@ -22,6 +22,9 @@
 [медиа-[а-я]]i
 hint="-> Слова медиа... пишутся слитно"
 
+[спец-[а-я]]i
+hint="-> Слова спец... пишутся слитно"
+
 [аудио файл]i
 hint="-> Слова аудио... пишутся слитно"
 
@@ -38,7 +41,7 @@
 valid msgstr="^Сопровождающий$"
 hint="Maintainer -- Сопровождающий"
 
-[&[a-zA-Z]]
+[---&(?=[a-zA-Z])] # Акселераторов в тексте нет
 valid before="[a-z]*="
 valid before="(nbsp|amp|reg|lt|gt|quot|deg|ge|ndash|copy|auml|mdash);"
 valid msgid="Drag&Drop"
@@ -68,15 +71,19 @@
 
 [  ]
 valid msgid="  "
+valid srcref="\(programlisting\)"
+valid srcref="\(synopsis\)"
+valid after="<synopsis.*" before=".*?</synopsis"
+valid after="<screen.*" before=".*?</screen"
 hint="Уберите двойной пробел" # FIXME: not for docs!
 
-[колон(к|ок)]i
+[-- колон(к|ок)]i # TODO: Этим нужно заниматься дополнительно, а пока игнорировать
 hint="column -- столбец"
 
 [в течении]i
 hint="«в течении» -- разве что «в течении реки»"
 
-[перезаписать]i
+[-- перезаписать]i # LAW: не всегда ("в зависимости от параметра может потребоваться перезаписать таблицы...")
 hint="«overwrite» переводится как «заменить»"
 
 [настройки]i
@@ -84,39 +91,39 @@
 valid after="средства "
 valid after="кнопку "
 valid after="при ошибке установки или "
-valid after="мастера "
+valid after="мастер(а)? "
 valid after="Центр "
 valid after="утилиты "
 valid after="возможность "
 valid after="изменения "
 valid after="Программа "
 valid after="программу "
-valid after="окне "
-valid after="окно "
-valid after="окна "
-valid after="файл "
+valid after="окн[аое] "
+valid after="файл(е)? "
 valid after="вашей "
-valid after="параметры "
 valid after="сценарий "
 valid after="ошибка "
-valid after="диалога "
-valid after="диалог "
-valid after="модуля "
-valid after="модуле "
+valid after="диалог(а|е)? "
 valid after="возможности "
-valid after="модуль "
-valid after="модули "
+valid after="модул(e|и|ь|я) "
 valid after="для "
-valid after="мастер "
 valid after="неправильной "
 valid after="быстрой "
 valid after="автоматической "
-valid after="диалоге "
-valid after="параметрах "
-valid after="файле "
+valid after="параметр(а|ах|ы)? "
 valid after="завершения "
 valid after="экрана "
 valid after="модулям "
+valid after="аспектов "
+valid after="обслуживания (и|или) "
+valid after="правило "
+valid after="правильности "
+valid after="дополнительной "
+valid after="начальной "
+valid after="упрощения "
+valid after="процедура "
+valid after="способ(а|ы) "
+valid after="время "
 hint="There's no such plural form 'настройки', use 'параметры' or something else. If you see this message but there is only 'настройки' as the genitive case of 'настройка', please, report it."
 
 [настройках]i
@@ -146,7 +153,7 @@
 [знач[её]к]i
 hint="орфография"
 
-[горяч]i
+[-- горяч]i # У нас это ещё и горячий резерв
 hint="keyboard shortcut, hotkey -- комбинация клавиш"
 
 [ярлык]i
@@ -155,6 +162,8 @@
 
 [скач]i
 valid after="^Са"
+valid after="резком "
+valid before="ков"
 hint="не используем слова скачивать, скачивание и т.д."
 
 [рассклад]i
@@ -181,8 +190,23 @@
 valid before="\">"
 valid after="=&"
 valid after="googl" before="\.com"
+valid after="<screen.*" before=".*?</screen"
+valid srcref="\(literal\)"
+valid srcref="\(programlisting\)"
+valid after="<literal .*" before=".*?</literal>"
 hint="слово email нужно переводить"
 
+[fork]i
+valid srcref="\(parameter\)" msgid="^fork$"
+valid before="Number"
+hint="слово fork нужно переводить"
+
+#[filenode]i
+#valid after="_"
+#valid after="rel"
+#valid ctx="parameter"
+#hint="слово filenode нужно переводить"
+
 [не подписанн]i
 hint="'неподписанный' -- слитно (чаще всего)"
 
@@ -225,6 +249,12 @@
 valid msgid="\(# will be replaced by ascending numbers\)" msgstr="\(символ # будет заменён порядковым номером\)" cat="dolphin" # '#' is hardcoded in Dolphin
 valid msgstr="«#»" # FIXME: "before" or "after"?
 valid msgstr="^\*99#$"
+valid after="<screen.*" before=".*?</screen"
+valid after="<literallayout.*" before=".*?</literallayout"
+valid after="<literal.*" before=".*?</literal"
+valid srcref="\(literal\)"
+valid srcref="\(programlisting\)"
+valid msgid="comment"
 hint="заменить на '№'?"
 
 [\bне не\b]i
@@ -259,6 +289,15 @@
 hint="shortcut -- комбинация клавиш"
 
 [superuser]i
+valid srcref="\(synopsis\)"
+valid srcref="\(literal\)"
+valid after="<literal.*" before=".*?</literal"
+valid after="<computeroutput.*" before=".*?</computeroutput"
+valid before="_reserved_connections"
+valid after="<quote>" before="</quote>"
+valid after="guc-"
+valid after="--"
+valid after="--no-"
 hint="superuser нужно перевести"
 
 [\bклик]i
@@ -280,7 +319,7 @@
 [проверьте вашу установку]i
 hint="не по-русски"
 
-[\bok\b]i
+[-- bok\b]i # Не перевод GUI -- не актуально
 hint="Ok -> ОК (русскими буквами)"
 
 [[а-яіїѣѳўє][a-z]]i
@@ -294,9 +333,11 @@
 hint="Используйте специальный символ ©"
 
 [copyright]i
+valid srcref="\(literal\)"
 hint="Используйте специальный символ ©"
 
 [\bне могу\b]i
+valid after="<quote.*" before=".*?</quote"
 hint="-> 'не удалось'"
 
 [mime]i
@@ -314,12 +355,20 @@
 
 [;']
 valid after="[0-9]px" # example: <h2 style='margin-top: 0px;'>
+valid after="(&zwsp|&nbsp|&gt|&amp)"
+valid after="<screen.*" before=".*?</screen"
+valid after="<literal.*" before=".*?</literal"
+valid srcref="\(programlisting\)"
 hint="Не стоит изобретать падежи у несклоняемых названий"
 
-[ ,]
+[\s,]
+valid after="<synopsis.*" before=".*?</synopsis"
+valid msgstr="<placeholder-1/> <placeholder-2/> ,"
+valid srcref="\(synopsis\)"
+valid srcref="\(programlisting\)"
 hint="Уберите пробел перед запятой"
 
-[\.[а-я]]i
+[-- \.[а-я]]i # Некорректное правило
 valid after="и т" span="д" before="\." # "и т.д." # некоторые говорят, что "и т.д." нужно писать полностью ("и так далее")
 valid after="и т" span="п" before="\." # "и т.п."
 valid after="т" span="к" before="\." # "т.к."
@@ -329,7 +378,16 @@
 valid after="дюймов рт" span="с" before="т\." # дюймов рт.ст.
 valid after="дюймы рт" span="с" before="т\." # дюймы рт.ст.
 valid after="\.\." span="\.ч" before="то"
+
+[\.(?=[а-я])]i
 valid after="@\*" # маска пользователя в Konversation
+valid srcref="\(programlisting\)"
+valid after="имя_таблицы" before="имя_колонки"
+valid after="схема" before="таблица"
+valid after="\*" before="таблица"
+valid after="отношение" before="атрибут"
+valid after="табл1" before="атр1"
+valid after="табл2" before="атр2"
 hint="Поставьте пробел после точки в конце предложения"
 
 [директори]i
@@ -388,7 +446,11 @@
 hint="В кнопке 'Отмена' акселератор ставьте на 'т'"
 
 [ \.\.\.$]
-hint="В командах меню перед многоточием не ставьте пробел."
+valid srcref="\(optional\)" msgid=" \.\.\.$"
+valid srcref="\(literal\)" msgid=" \.\.\.$"
+valid srcref="\(programlisting\)" msgid=" \.\.\.$"
+valid after=","
+hint="Перед многоточием не ставьте пробел."
 
 [«спам]i
 hint="'спам' должно быть без кавычек"
@@ -409,6 +471,9 @@
 hint="'невисокосный' пишется слитно"
 
 [ввиду]i
+valid before=" того"
+valid before=" такой"
+valid after="особенно "
 hint="В контексте 'Имейте в виду, что ...', 'в виду' пишется раздельно. "
 
 [выберете]i
@@ -416,6 +481,7 @@
 hint="В повелительном наклонении должно быть 'выберите'"
 
 [параграф]i
+valid msgid="paragraph and the following two paragraphs appear in all copies"
 valid !msgid="paragraph"
 hint="paragraph -- абзац"
 
@@ -438,32 +504,52 @@
 [на брауз]i
 hint="на браузерах -> в браузерах"
 
-[(Вам|Вы|Ваш|Вашей)\b]
+[(?<!^)(Вам|Вы|Ваш|Вашей)\b]
 valid after="\.\s+"
-valid after="<[a-z]+>"
+valid after="\.\)\s+"
+valid after="<[a-z\-_0-9/]+>\s*"
 hint="Пишем 'вы' со строчной буквы."
 
-[отображ]i
+[-- отображ]i # Неактуальное правило, у нас есть перевод "map" -> "отображать"
 hint="Избегаем использования слова 'отображать' и однокоренных в смысле показа чего-либо на экране, вместо этого пишем 'показывать'"
 
 [иметь эффект]i
 hint="takes effect -> вступает в силу"
 
-[[^\.]\. [а-я]]
-valid after="э" span="л. п" before="очты" # частое сокращение "эл. почты"
+[(?<=[^\.])\.\s(?=[а-я])]
+valid after="эл" before="почты" # частое сокращение "эл. почты"
+valid after="\bсм"
+valid after="\bСм"
+valid after="[0-9]+ г"
+valid after="[тТ]. к"
+valid after="[тТ]" before="к."
+valid after="\bи т. д"
+valid after="\bи т" before="д\."
+valid after="\bи т. п"
+valid after="\bи т" before="п\."
+valid after="\bн. э"
+valid after="\bн" before="э\."
+valid after="[тТ]. е"
+valid after="\b[тТ]" before="е\."
+valid after="\bмат" before="представлен"
+valid after="в стиле #"
 hint="Предложения пишутся с заглавной буквы"
 
 [на строке]
 hint="-> в строке"
 
+[на таблице]
+hint="-> в таблице"
+
+
 # Some people "invented" separating keywords by semicolons, find them
 [[^а-яёa-z0-9\ ,«»\-\/\+]]i
 valid !ctx="^X-KDE-Keywords$"
 hint="X-KDE-Keywords should contain words separated by commas"
 
-[\!\s]
+[-- \!\s] # В нашей документации допустимо
 hint="Восклицания допустимы только в играх и программах для детей"
-[\!$]
+[-- \!$] # В нашей документации допустимо
 hint="Восклицания допустимы только в играх и программах для детей"
 
 [конверти]i
@@ -488,9 +574,10 @@
 #   градуса, минуты и секунды от цифр не отбивают. Знак °С отбивают, как и
 #   другие обозначения единиц физ. величин, на 2 п.
 # 2. http://habrahabr.ru/post/23250/
-[%.%]
+[-- %.%] # LAW: https://ru.wikipedia.org/wiki/Википедия:Отбивка_знака_процента_от_предшествующей_цифры
 hint="Знак процента отделяют от числа коротким пробелом (каким именно -- читайте на l10n.lrn.ru)"
-[[0-9]%]
+
+[-- [0-9]%] # LAW: https://ru.wikipedia.org/wiki/Википедия:Отбивка_знака_процента_от_предшествующей_цифры
 hint="Знак процента отделяют от числа коротким пробелом (каким именно -- читайте на l10n.lrn.ru)"
 
 # http://lists.kde.ru/pipermail/kde-russian/2008-July/012291.html
@@ -501,6 +588,12 @@
 [\n]
 valid msgid="\n"
 valid ctx="(qtundo-format)"
+valid after="<screen.*" before=".*?</screen"
+valid srcref="\(programlisting\)"
+valid srcref="\(synopsis\)"
+valid after="<synopsis.*" before=".*?</synopsis"
+valid srcref="\(screen\)"
+valid srcref="\(entry\)"
 hint="No '(qtundo-format)' in context string, hence magic split is not available"
 
 # https://ru.wikipedia.org/wiki/%D0%9C%D0%B0%D0%BB%D0%B0%D1%8F%D0%BB%D0%B0%D0%BC_(%D0%BF%D0%B8%D1%81%D1%8C%D0%BC%D0%BE)
@@ -513,3 +606,230 @@
 
 # TBD: название программы "Параметры системы" склоняется? Например: "в Параметрах системы"?
 
+
+[::]
+valid before="(&zwsp;)?[a-zA-Z]"
+valid before="(/)?[0-9]"
+valid before="<replaceable"
+valid msgid="^::$"
+valid before="</(literal|primary)"
+valid srcref="\(programlisting\)"
+hint="Двойные двоеточия"
+
+[(?<=[а-я\.])\(]i
+valid before="(а|и|ы|ов|ам|ей|ами|я|ым)\)"
+valid before="(Y|M|D)\)"
+valid after="имя-функции"
+valid after="функция_массив"
+valid after="функция_кортеж"
+hint="Не хватает пробела перед скобкой."
+
+[>[а-я]]i
+valid after="<(replaceable|parameter|type|optional)[^>]*"
+valid after="<(quote|citetitle)"
+valid after="<(firstterm|acronym|foreignphrase)"
+valid after="<lineannotation"
+valid after="<emphasis"
+valid after="<see"
+valid after="<seealso"
+valid after="<(primary|secondary|tertiary)[^>]*"
+valid after="<(link|ulink)[^>]+"
+valid after="<literal[^>]+"
+valid after="<(structname|varname|structfield)"
+valid after="<(screen|computeroutput)[^>]+"
+valid after="<application[^>]*"
+valid after="<placeholder-[0-9]+/"
+valid after="</screen"
+valid after="</synopsis"
+hint="Не хватает пробела или неправильный тег."
+
+# <emphasis> долю </emphasis>
+ 
+[[а-я]<(?=[a-z])]i
+valid before="placeholder-1/>$" # предполагается, что в placeholder-1 сидит тег primary
+valid before="placeholder-1/>," # предполагается, что в placeholder-1 сидит тег primary
+hint="Не хватает пробела или неправильный тег."
+
+[[а-я]\.\.($|[^.])]
+hint="Две точки."
+
+[[а-я>] \.($|[^.])]
+hint="Пробел перед точкой."
+
+[\s:($|[^:^=])]
+valid after="<synopsis.*" before=".*?</synopsis"
+valid after="<screen.*" before=".*?</screen"
+valid after="<literal.*" before=".*?</literal"
+valid msgstr="<placeholder-1/> <placeholder-2/> ,"
+valid srcref="\(synopsis\)"
+valid srcref="\(programlisting\)"
+valid after="set"
+hint="Уберите пробел перед двоеточием"
+
+[/ >]
+hint="Пробел в закрывающемся теге."
+
+[\bor\b]
+valid after="<screen.*" before=".*?</screen"
+valid after="<computeroutput.*" before=".*?</computeroutput"
+hint="Слово or надо переводить."
+
+[\bпо\s[а-я]+му(?=$|\.|,)]i
+valid span="по одному"
+valid span="по нему"
+valid span="по всему"
+hint="Должно быть 'по-...'?"
+
+[, и т\. д\.]
+hint="Запятая перед 'и т. д.'"
+
+[и\s?тд\.]
+hint="Должно быть 'и т. д.'"
+
+[в\sпоследствии]i
+hint="Должно быть 'впоследствии'"
+
+[ --? ]
+valid srcref="\(literal\)"
+valid srcref="\(programlisting\)"
+valid srcref="\(screen\)"
+valid after="<literal.*" before=".*?</literal"
+valid after="<screen.*" before=".*?</screen"
+valid after="<synopsis.*" before=".*?</synopsis"
+valid before="[0-9]"
+hint="Должно быть тире (&mdash;)."
+
+[здесь(:)? <xref]i
+hint="Надо заменить на 'в <xref...' или что-то подобное."
+
+[смотри\b]i
+hint="Должно быть см. или смотрите."
+
+[\sв\s<xref(?![^>]+remap=\s*"6").*?/>]i # Предложный падёж
+valid !after="(\
+говорилось|\
+дано|\
+найти|\
+об этом|\
+обзору|\
+обсуждается|\
+обсуждалось|\
+объясняется|\
+описан|\
+описана|\
+описание|\
+описанная|\
+описанному|\
+описанные|\
+описано|\
+описаны|\
+описывалось|\
+описываются|\
+определяется|\
+перечислены|\
+подробнее|\
+подробно|\
+подробности|\
+показан|\
+показано|\
+представлен|\
+приведена|\
+приведено|\
+приведены|\
+приведённым|\
+приведённому|\
+пример|\
+рассказывается|\
+рассматривается|\
+рассмотренным|\
+рассмотрены|\
+сказано|\
+увидеть|\
+упомянуты выше\
+)"
+hint="Не хватает правильного падежа во фразе \"в <xref\" (должен быть remap=\"6\"?)."
+
+[\sк\s<xref(?![^>]+remap=\s*"3").*?/>]i # Дательный падёж
+hint="Не хватает правильного падежа во фразе \"к <xref\" (должен быть remap=\"3\"?)."
+
+[\bсм\.\s<xref(?![^>]+remap=\s*"4").*?/>]i # Винительный падёж
+hint="Не хватает правильного падежа во фразе \"см. <xref\" (должен быть remap=\"4\"?)."
+
+[\bсм\.\sв\s]i
+hint="Надо заменить 'см. в' на 'описано в' или что-то подобное."
+
+[информация (появится|находится) в]i
+hint="Надо заменить на '... описано в/За подробностями обратитесь к ' или что-то подобное."
+
+[\bIP(</[a-z]+>)?\s(адрес|соединен|подключен)]i
+hint="IP — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bIPv[46](</[a-z]+>)?\s(адрес|соединен)]i
+hint="IPv4, IPv6 — не прилагательные, надо писать через дефис или перефразировать."
+
+[TCP(</[a-z]+>)?\s(порт)]i
+hint="TCP — не прилагательное, надо писать через дефис или перефразировать."
+
+[ident(</[a-z]+>)?\s(аутентифик|сервер)]i
+hint="ident — не прилагательное, надо писать через дефис или перефразировать."
+
+[peer(</[a-z]+>)?\s(аутентифик)]i
+hint="peer — не прилагательное, надо писать через дефис или перефразировать."
+
+[LDAP(</[a-z]+>)?\s(аутентифик|сервер|подключен|URL)]i
+hint="LDAP — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bRADIUS(</[a-z]+>)?\s(сервер|аутентифик)]i
+hint="RADIUS — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bURL(</[a-z]+>)?\s(схема)]i
+hint="URL — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bUnix(</[a-z]+>)?\s(систем|сервер|сокет|утилит|домен)]i
+hint="Unix — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bSQL(</[a-z]+>)?\s(скрипт|оператор|команд|объект|запрос)]i
+valid after="соответствующей стандарту "
+valid after="В стандарте "
+valid after="В " span="SQL запрос" before="ы"
+hint="SQL — не прилагательное, надо писать через дефис или перефразировать."
+
+[pgSQL(</[a-z]+>)?\s(скрипт|оператор|команд|объект|функц|переменн)]i
+hint="pgSQL — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bDDL(</[a-z]+>)?\s(оператор|команд|объект)]i
+hint="DDL — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bWAL(</[a-z]+>)?\s(файл|архив|запис|сегмент|лог|данн)]i
+valid span="WAL запис" before="(ан|ы)"
+hint="WAL — не прилагательное, надо писать через дефис или перефразировать."
+
+[\bTOAST(</[a-z]+>)?\s(табл|столб)]i
+hint="TOAST — не прилагательное, надо писать через дефис или перефразировать."
+
+
+#TODO: [<(primary|secondary|tertiary)>\s*[a-z]]i
+# hint="Термин для индекса надо переводить."
+
+[(?<!,)\s+чтобы]
+valid after=";"
+valid after="(а|но|то|и|И)"
+valid after="(просто|только)"
+valid after="([Дд]ля того|Вместо того|с тем)"
+hint="Запятая перед 'чтобы ...'"
+
+[\bsee\b]i
+valid after="<" before=">"
+valid after="</" before=">"
+hint="See надо переводить"
+
+# [\sand\s] # Требуется уточнить правило для постоянного включения
+# hint="Не переведено and."
+
+# [\b[tT]he\s] # Требуется уточнить правило для постоянного включения
+# hint="Не переведено the."
+
+# [\bfor\s] # Требуется уточнить правило для постоянного включения
+# hint="Не переведено for."
+
+# TODO: поменять <quote>...</quote> на &laquo;...&raquo;, чтобы были кавычки-ёлочки?
diff -u -r -N pology-0.12.ori/lang/ru/rules/letter-yo.rules pology-0.12/lang/ru/rules/letter-yo.rules
--- pology-0.12.ori/lang/ru/rules/letter-yo.rules	2014-07-24 22:57:17.000000000 +0400
+++ pology-0.12/lang/ru/rules/letter-yo.rules	2015-12-11 13:21:21.227585644 +0300
@@ -13,6 +13,13 @@
 valid before="и"
 hint="Буква ё"
 
+[заключен]i
+valid before="а"
+valid before="ы"
+valid before="о"
+valid before="и"
+hint="Буква ё"
+
 [отключен]i
 valid before="а"
 valid before="ы"
@@ -34,16 +41,46 @@
 valid before="и"
 hint="Буква ё"
 
-[учетн]i
+[чет]i
+valid after="хо"
+valid after="со"
+valid after="с" before="[ао]"
+valid after="вы"
+valid before="в[её]р"
+valid before="ыр"
 hint="Буква ё"
 
 [ведет]i
+valid after="вы" before="\s"
+hint="Буква ё"
+
+[дает]i
+valid after="па"
+valid after="блю"
+valid after="обла"
+valid after="рож"
+valid after="преж"
+valid after="ожи"
+valid after="суж"
+valid after="тверж"
+valid after="вреж"
+valid after="бож"
+valid after="вож"
+valid after="беж"
+valid after="нуж"
+valid after="стра"
 hint="Буква ё"
 
 [звезд[ н]]i
 hint="Буква ё"
 
-[ ee]i # how to make this more common (add a \b in the beginning?)
+[\bee\b]i # how to make this more common (add a \b in the beginning?)
+hint="Буква ё"
+
+[\bее\b]i
+hint="Буква ё"
+
+[\bнее\b]
 hint="Буква ё"
 
 [щелкните]i
@@ -52,12 +89,187 @@
 [\bнем\b]
 hint="Буква ё"
 
-[\bотклонен\b]i
+[\bеще\b]i
+hint="Буква ё"
+
+[клоненн]i
+hint="Буква ё"
+
+[клонен\b]i
+valid="с"
+hint="Буква ё"
+
+[несенн]i
+valid after="вы"
+hint="Буква ё"
+
+[несен\b]i
+valid after="вы"
+hint="Буква ё"
+
+[ъем]i
+valid before="лем"
+hint="Буква ё"
+
+[жестк]i
+hint="Буква ё"
+
+[\bимен\b]i
+hint="Буква ё"
+
+[\bсерьез]i
+hint="Буква ё"
+
+[веденн]
+valid after="вы"
+hint="Буква ё"
+
+[веден\b]
+valid after="вы"
+hint="Буква ё"
+
+[деленн]
+valid after="вы"
+hint="Буква ё"
+
+[делен\b]
+valid after="вы"
+hint="Буква ё"
+
+[даленн]
+hint="Буква ё"
+
+[дален\b]
+hint="Буква ё"
+
+[жденн]
+valid after="ну"
+hint="Буква ё"
+
+[жден\b]
+valid after="ну"
+hint="Буква ё"
+
+[жненн] # осложнённый
+hint="Буква ё"
+
+[жнен\b]
+hint="Буква ё"
+
+[ененн] # изменённый
+hint="Буква ё"
+
+[енен\b] # изменён
+valid after="бесц"
+hint="Буква ё"
+
+[епленн] # закрепленный
+hint="Буква ё"
+
+[еплен\b] # закреплен
+hint="Буква ё"
+
+[ещенн]
+hint="Буква ё"
+
+[ещен\b]
+hint="Буква ё"
+
+[еченн]
+valid after="м"
+hint="Буква ё"
+
+[ечен\b]
+valid after="м"
+hint="Буква ё"
+
+[ищенн] # защищённый
+valid after="оч"
+hint="Буква ё"
+
+[ищен\b]
+valid after="оч"
+hint="Буква ё"
+
+[ащенн] # сокращённый
+hint="Буква ё"
+
+[ащен\b] # сокращён
+hint="Буква ё"
+
+[аненн] # распространённый
+hint="Буква ё"
+
+[анен\b] # распространён
+hint="Буква ё"
+
+[верты] # развёртывание
+hint="Буква ё"
+
+[вернет]
+hint="Буква ё"
+
+[путем\b]
+hint="Буква ё"
+
+[надежн]
+hint="Буква ё"
+
+[дешев]
+valid before="ле"
+hint="Буква ё"
+
+[черки]
+hint="Буква ё"
+
+[ойдет\b]
+hint="Буква ё"
+
+[\bоберт]
+hint="Буква ё"
+
+[рвет]
+hint="Буква ё"
+
+[придет]
+hint="Буква ё"
+
+[чье]
+hint="Буква ё"
+
+[вершенн] # завершённый
+valid before="о\b"
+hint="Буква ё"
+
+[вершен\b] # завершён
+hint="Буква ё"
+
+[единенн] # соединённый
+hint="Буква ё"
+
+[единен\b]
+hint="Буква ё"
+
+[угленн] # округлённо
+hint="Буква ё"
+
+[углен\b] # округлён
+hint="Буква ё"
+
+[ражен\b] # отображён
+hint="Буква ё"
+
+[раженн\b] # отображёный
+hint="Буква ё"
+
+[стает] # остаётся
+valid after="зра"
 hint="Буква ё"
 
-[\bудален\b]i
+[берет]
+valid after="вы"
 hint="Буква ё"
 
-[\bперенесен\b]i
+[[зс]вернут[^ь]]
 hint="Буква ё"
 
diff -u -r -N pology-0.12.ori/lang/ru/spell/postgres.aspell pology-0.12/lang/ru/spell/postgres.aspell
--- pology-0.12.ori/lang/ru/spell/postgres.aspell	1970-01-01 03:00:00.000000000 +0300
+++ pology-0.12/lang/ru/spell/postgres.aspell	2015-12-19 08:10:55.161481272 +0300
@@ -0,0 +1,346 @@
+personal_ws-1.1 ru 0 UTF-8
+Вуаля
+ГГГГММДД
+ГГММДД
+ДАЁТ
+НЕСЁТ
+ОРСУБД
+РСУБД
+УВЕДОМЛЁН
+ЦС
+ЧЧММ
+ЧЧММСС
+Южноавстралийского
+Юникод
+Юникоде
+автонастройка
+автоочистка
+автоочистке
+автоочистки
+автоочисткой
+автоочистку
+автоувеличением
+автофиксацией
+автофиксацию
+автофиксация
+амортизироваться
+атомарностью
+беззнаковыми
+битностью
+булевого
+буферы
+версионности
+взаимоблокировка
+взаимоблокировки
+взаимоблокировку
+взаимоблокировок
+взаимозаменяем
+восьмибайтное
+восьмибайтном
+восьмибайтные
+восьмибайтных
+восьмибайтовые
+восьмибайтовых
+восьмизначной
+восьмизначные
+гипотезирующая
+гипотезирующей
+гипотезирующие
+гипотезирующий
+гипотезирующих
+двумерного
+двумерном
+двумерным
+двумерных
+двусторонняя
+двухмерный
+двухфазного
+двухфазное
+двухфазной
+двухфазные
+двухфазный
+декодироваться
+декомпиляции
+демоверсия
+журналировании
+журналируемая
+журналируемые
+журналируемыми
+заархивировалась
+заархивируется
+заархивируются
+заблокировался
+заблокируются
+задействуется
+задействуются
+задействующим
+задействуя
+закешированные
+запустится
+запуститься
+затребуется
+зацикливается
+зацикливались
+зацикливаний
+зацикливаться
+инкапсулирующий
+интерполируя
+интрасети
+исследуются
+исследовалась
+категоризируются
+контрпримера
+криптосоли
+локалью
+локалями
+локалях
+местозаполнители
+метакоманд
+метакоманда
+метакомандами
+метакомандах
+метакоманде
+метакомандой
+метакоманду
+метакоманды
+метасимволов
+метасимволом
+метасимволы
+метасинтаксис
+метасинтаксиса
+метастраница
+метастраницах
+минидамп
+многоверсионного
+многоверсионное
+многоверсионность
+многоверсионности
+многопоточности
+многопоточность
+многосимвольное
+моделируется
+модифицируется
+модифицируются
+моноширинным
+мультитранзакции
+мультитранзакций
+мультитранзакция
+мультитранзакциях
+мультиязычный
+неактивности
+неанонимного
+незаархивированные
+неархивированных
+неблокирующий
+неблокирующим
+неблокирующих
+негруппируемая
+негруппируемой
+негруппируемую
+негруппируемым
+недетерменированности
+недоверенного
+недоверенный
+недоверенным
+недоверенными
+недопустим
+нежурналируемая
+нежурналируемой
+нежурналируемые
+нежурналируемыми
+нежурналируемых
+неинтуитивным
+неконфликтующие
+нелинейны
+неоткладываемого
+неоткладываемое
+неоткладываемое
+неоткладываемой
+неоткладываемые
+неотсечённую
+неотсечённый
+непересечения
+непредсказуема
+неприменим
+неприхотлив
+нескалярных
+несканируемом
+несовместим
+несравниваемый
+одинаков
+одномерного
+одномерные
+одномерный
+одномерным
+одномерных
+однострочном
+однотипно
+однотипных
+округлённо
+перезаписывались
+перезаписываются
+переиндексации
+переиндексацию
+переиндексация
+переиндексируется
+перекомпиляции
+переконфигурировать
+переместительным
+переподключиться
+пересоберите
+пересобирались
+пересобирать
+пересобрав
+пересобрать
+перестраивания
+перцентили
+перцентиль
+перцентилю
+перцентиля
+платформозависимая
+платформозависимыми
+поддиапазонов
+подзапрос
+подзапроса
+подзапросам
+подзапросами
+подзапросах
+подзапросе
+подзапросов
+подзапросом
+подзапросу
+подзапросы
+подкатегорий
+подмассивы
+подпроцесс
+подпроцессом
+подтранзакции
+подтранзакций
+подтранзакциях
+полужирным
+портировании
+портирования
+портированный
+портировать
+портируемости
+портируя
+потокобезопасное
+предзагрузка
+предсказуем
+преобразуемое
+преобразуемые
+преобразующей
+преобразующие
+преобразующими
+преобразующих
+приемлем
+пролептический
+пролистывать
+просканировав
+просканировал
+просканирован
+просканированного
+просканированную
+просканированных
+просканированы
+прототипирования
+псевдобазе
+псевдоотношения
+псевдоотношениям
+псевдороль
+псевдотип
+псевдотипа
+псевдотипами
+псевдотипов
+псевдотипы
+псевдоуникальный
+пятисимвольное
+пятисимвольный
+разблокируется
+размечивания
+регистрозависимы
+регистронезависимый
+регистронезависимым
+регистронезависимыми
+ресурсоёмких
+самоисключающим
+самоопределяемых
+самопересечением
+самоподписанного
+самоподписанный
+сериализация
+сериализовать
+сериализуемая
+сериализуемого
+сериализуемой
+сериализуемости
+сериализуемость
+сериализуемые
+сериализуемый
+сериализуемыми
+сериализуемых
+сканирований
+сканированиям
+сканированиях
+словосоставляющий
+словосоставляющих
+соптимизированы
+соптимизировать
+сортировались
+сортироваться
+сортируется
+сортируются
+спецкодов
+спецкоды
+спецобозначением
+спецобозначения
+спецпоследовательностей
+спецпоследовательности
+спецпоследовательность
+спецпоследовательностью
+спецпоследовательностями
+спецпоследовательностях
+спецстрок
+спецстроках
+спецстроке
+спецстроки
+спецустройств
+спецфайле
+стеммер
+стеммером
+стеммеру
+стеммеры
+стемминга
+суперпользователей
+суперпользователем
+суперпользователи
+суперпользователь
+суперпользователю
+суперпользователя
+суперпользователям
+типобезопасность
+типобезопасные
+триггер
+триггеры
+трём
+трёхзначного
+трёхзначное
+трёхкратное
+удобоварим
+ужесточатся
+хешированное
+хешированный
+хешированных
+хешируется
+центральноевропейские
+четырёхбайтное
+четырёхбайтового
+четырёхбайтовое
+четырёхбайтовым
+четырёхбайтовых
+четырёхзначного
+четырёхзначные
+шестизначной
+шестизначные
+южноевропейские
+языкозависимые
diff -u -r -N pology-0.12.ori/lang/ru/spell/postgres-names.aspell pology-0.12/lang/ru/spell/postgres-names.aspell
--- pology-0.12.ori/lang/ru/spell/postgres-names.aspell	1970-01-01 03:00:00.000000000 +0300
+++ pology-0.12/lang/ru/spell/postgres-names.aspell	2015-11-25 17:40:55.000000000 +0300
@@ -0,0 +1,12 @@
+personal_ws-1.1 ru 0 UTF-8
+Стоунбрейкером
+Хейуорд
+Хейуорду
+Хейуорде
+Уолли
+д'Арк
+Кормаком
+Тадхоуп
+Хангыль
+Скалигера
+Тридентского
diff -u -r -N pology-0.12.ori/pology/markup.py pology-0.12/pology/markup.py
--- pology-0.12.ori/pology/markup.py	2014-07-24 22:57:17.000000000 +0400
+++ pology-0.12/pology/markup.py	2015-12-01 11:52:48.218495680 +0300
@@ -1417,7 +1417,7 @@
 
         if (   msg.msgctxt in ignctxt
             or msg.msgid in ignid
-            or (msg.msgctxt is not None and msg.msgctxt.startswith(ignctxt))
+#            or (msg.msgctxt is not None and msg.msgctxt.startswith(ignctxt))
             or msg.msgid.startswith(ignidsw)
         ):
             return [] if spanrep else 0
@@ -2027,7 +2027,7 @@
 
     missing_plnums = list(orig_plnums.difference(trans_plnums))
     extra_plnums = list(trans_plnums.difference(orig_plnums))
-    if missing_plnums:
+    if missing_plnums and False:
         tags = "".join(["<placeholder-%s/>" % x for x in missing_plnums])
         errmsg = _("@info",
                    "Missing placeholder tags in translation: %(taglist)s.",
diff -u -r -N pology-0.12.ori/pology/remove_ru.py pology-0.12/pology/remove_ru.py
--- pology-0.12.ori/pology/remove_ru.py	1970-01-01 03:00:00.000000000 +0300
+++ pology-0.12/pology/remove_ru.py	2015-11-25 17:41:32.000000000 +0300
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+"""
+Remove russian specific substrings from text.
+
+@author: Alexander Lakhin <exclusion@gmail.com>
+@license: GPLv3
+"""
+
+import re
+
+_g_re = re.compile(ur"([0-9]{4})\sг\.", re.U)
+_num_re = re.compile(ur"([0-9]+)-(ый|ую)", re.U)
+_td_re = re.compile(ur"\bт\. д\.", re.U)
+_tp_re = re.compile(ur"\bт\. п\.", re.U)
+_te_re = re.compile(ur"\bт\. е\.", re.U)
+_nd_re = re.compile(ur"\bн/д\b", re.U)
+
+_email_re = re.compile(ur"\sэл\.( почт)", re.U)
+_plural_ending_re = re.compile(ur"(\w)(\(ы\)|\(ым\)|\(ов\))", re.U)
+
+def remove_abbr(msgstr, msg, cat):
+    """
+    [type F3C hook]
+    """
+    msgstr = _g_re.sub(ur"\1", msgstr)
+    msgstr = _num_re.sub(ur"\1", msgstr)
+    msgstr = _email_re.sub(ur"\1", msgstr)
+    msgstr = _td_re.sub(ur"так далее", msgstr)
+    msgstr = _tp_re.sub(ur"тому подобное", msgstr)
+    msgstr = _te_re.sub(ur"то есть", msgstr)
+    msgstr = _nd_re.sub(ur"нет данных", msgstr)
+    return msgstr
+
+
+_valid_re=re.compile(ur"\b(каким-нибудь|какой-нибудь|кому-нибудь|16-ричном|16-ричный|Спецпоследова\-\sтельность)\b", re.U)
+_valid_identifiers=re.compile(ur"("\
+u"число_вхожд|"\
+u"функция_состояния_движ|"\
+u"функции_состояния_движ|"\
+u"функция_завершения_движ|"\
+u"начальное_условие_движ|"\
+u"обратная_функция_движ|"\
+u"тип_данных_состояния_движ|"\
+u"размер_данных_состояния_движ|"\
+u"сигнатура_агр_функции|"\
+u"строк_в_результате|"\
+u"коммут_оператор|"\
+u"функция_лекс_типов|"\
+u"внутр_длина|"\
+u"атр1|"\
+u"атр2|"\
+")", re.U)
+def validwords(msgstr, msg, cat):
+    """
+    [type F3C hook]
+    """
+    msgstr = _valid_identifiers.sub(ur"", msgstr)
+    msgstr = _valid_re.sub(ur"", msgstr)
+    return msgstr
+
+def remove_endings(msgstr, msg, cat):
+    msgstr = _plural_ending_re.sub(ur"\1", msgstr)
+    return msgstr
+
+_underscore_re = re.compile(ur"([а-я])_([а-я])", re.U + re.I)
+_zwsp_re = re.compile(ur"&zwsp;", re.U)
+def remove_underscore(msgstr, msg, cat):
+    msgstr = _underscore_re.sub(ur"\1 \2", msgstr)
+    msgstr = _zwsp_re.sub(ur"", msgstr)
+
+    return msgstr
+
+_people_re = re.compile(ur"("\
+u"Эндри Ю|"\
+u"Джолли Чен|"\
+u"Мартина Потера|"\
+u"Стефаном Симковичем|"\
+u"Георга Готлоба|"\
+u"Катрин Сейр|"\
+u"Жозефом Жюстом Скалигером|"\
+#u"|"\
+")", re.U)
+
+def remove_people(msgstr, msg, cat):
+    """
+    [type F3C hook]
+    """
+    msgstr = _people_re.sub("", msgstr)
+    return msgstr
